java/src/main/java/com/logicalclocks/hsfs/FeatureView.java

```diff
@@ -183,31 +185,34 @@ public List<List<Object>> previewFeatureVectors(Integer n, boolean external)
 
   @JsonIgnore
   public String getBatchQuery() throws FeatureStoreException, IOException, ParseException {
-    return getBatchQuery(null, null);
+    return getBatchQuery(null, null, null);
   }
 
   @JsonIgnore
-  public String getBatchQuery(String startTime, String endTime)
+  public String getBatchQuery(String startTime, String endTime, Integer trainingDataVersion)
```

Comment:
maybe a new method could be added that does not require `trainingDataVersion` (in a sense proxy method). This could be applicable in other places as well.